File: CS/WindowsApplication3/Main.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;


namespace DXSample {
    public partial class Main: XtraForm {
        public Main() {
            InitializeComponent();
        }

        GroupEditProvider provider;
        private void OnLoad(object sender, EventArgs e) {
            // TODO: This line of code loads data into the 'nwindDataSet.Products' table. You can move, or remove it, as needed.
            this.productsTableAdapter.Fill(this.nwindDataSet.Products);
            provider = new GroupEditProvider(gridView1);
            provider.ShowGroupEditorOnMouseHover = true;
            provider.SingleClick = true;
            provider.EnableGroupEditing();
        }

        protected override void OnFormClosing(FormClosingEventArgs e)
        {
            provider.DisableGroupEditing();
            base.OnFormClosing(e);
        }
    }
}

File: VB/WindowsApplication3/Main.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors


Namespace DXSample
	Partial Public Class Main
		Inherits XtraForm

		Public Sub New()
			InitializeComponent()
		End Sub

		Private provider As GroupEditProvider
		Private Overloads Sub OnLoad(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			' TODO: This line of code loads data into the 'nwindDataSet.Products' table. You can move, or remove it, as needed.
			Me.productsTableAdapter.Fill(Me.nwindDataSet.Products)
			provider = New GroupEditProvider(gridView1)
			provider.ShowGroupEditorOnMouseHover = True
			provider.SingleClick = True
			provider.EnableGroupEditing()
		End Sub

		Protected Overrides Sub OnFormClosing(ByVal e As FormClosingEventArgs)
			provider.DisableGroupEditing()
			MyBase.OnFormClosing(e)
		End Sub
	End Class
End Namespace
